bootstrap-pull-request/src/run.ts

```diff
@@ -0,0 +1,95 @@
+import * as core from '@actions/core'
+import * as github from '@actions/github'
+import * as git from './git'
+import { copyServicesFromPrebuilt } from './prebuilt'
+import { retryExponential } from './retry'
+import { writeNamespaceManifest } from './namespace'
+
+type Inputs = {
+  overlay: string
+  namespace: string
+  sourceRepository: string
+  destinationRepository: string
+  destinationRepositoryToken: string
+  namespaceManifest: string | undefined
+  substituteVariables: string[]
+}
+
+export const run = async (inputs: Inputs): Promise<void> =>
+  await retryExponential(() => bootstrapNamespace(inputs), {
+    maxAttempts: 50,
+    waitMs: 10000,
+  })
+
+const bootstrapNamespace = async (inputs: Inputs): Promise<void | Error> => {
+  core.info(`Checking out the prebuilt branch`)
+  const prebuiltDirectory = await checkoutPrebuiltBranch(inputs)
+  core.info(`Checking out the namespace branch`)
+  const namespaceDirectory = await checkoutNamespaceBranch(inputs)
+
+  const substituteVariables = parseSubstituteVariables(inputs.substituteVariables)
+  const [, sourceRepositoryName] = inputs.sourceRepository.split('/')
+
+  // TODO: consider the garbage collection of the outdated services
```

Comment:
Since this pull request is too large, I will add the garbage collection feature later.